lib/bot-service.js

```diff
@@ -26,7 +26,16 @@ botService.get = async (resource) => {
             });
             response.on('end', () => {
                 const data = decodeToken(body);
-                data === undefined ? reject() : resolve(data);
+                var responseBody = {};
+                if (data instanceof Error)  {
+                  reject(data)
+                } else if (response.statusCode !== 200){
```

Comment:
better to check 200 range instead of just 200. 202 accepted for instance

Reply:
Changed condition to > 202

Reply:
Also removed else blocks and changed var to let...